File: lib_elevator_control/src/main/java/com/reeman/agv/elevator/request/model/TaskInfo.java
package com.reeman.agv.elevator.request.model;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class TaskInfo implements Serializable {
    @SerializedName("passenger")
    public PassengerDTO passenger;
    @SerializedName("from")
    public FromDTO from;
    @SerializedName("to")
    public ToDTO to;
    @SerializedName("at")
    public String at;
    @SerializedName("in")
    public String in;
    @SerializedName("out")
    public Object out;
    @SerializedName("status")
    public Object status;

    public static class PassengerDTO {
        @SerializedName("id")
        public String id;
    }

    public static class FromDTO {
        @SerializedName("value")
        public Integer value;
    }

    public static class ToDTO {
        @SerializedName("value")
        public Integer value;
    }

    @Override
    public String toString() {
        return "TaskInfo{" +
                "passenger=" + passenger +
                ", from=" + from +
                ", to=" + to +
                ", at='" + at + '\'' +
                ", in='" + in + '\'' +
                ", out=" + out +
                ", status=" + status +
                '}';
    }
}

File: AGV_v1.9.0/lib_calling/src/main/java/com/reeman/agv/calling/model/TaskInfo.kt
package com.reeman.agv.calling.model

import com.google.gson.annotations.SerializedName

import android.os.Parcel
import android.os.Parcelable

class TaskInfo : Parcelable {
    @SerializedName("createTime")
    var createTime: Long = 0

    @SerializedName("startTime")
    var startTime: Long = 0

    @SerializedName("taskMode")
    var taskMode: Int = 0

    @SerializedName("token")
    var token: String? = null

    @SerializedName("targetPoint")
    var targetPoint: String = ""

    constructor()
    constructor(
        createTime: Long,
        startTime: Long,
        taskMode: Int,
        token: String?,
        targetPoint: String
    ) {
        this.createTime = createTime
        this.startTime = startTime
        this.taskMode = taskMode
        this.token = token
        this.targetPoint = targetPoint
    }

    companion object CREATOR : Parcelable.Creator<TaskInfo> {
        override fun createFromParcel(parcel: Parcel): TaskInfo {
            return TaskInfo(parcel)
        }

        override fun newArray(size: Int): Array<TaskInfo?> {
            return arrayOfNulls(size)
        }
    }

    private constructor(parcel: Parcel) {
        createTime = parcel.readLong()
        startTime = parcel.readLong()
        taskMode = parcel.readInt()
        token = parcel.readString()
        targetPoint = parcel.readString() ?: ""
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeLong(createTime)
        parcel.writeLong(startTime)
        parcel.writeInt(taskMode)
        parcel.writeString(token)
        parcel.writeString(targetPoint)
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun toString(): String {
        return "TaskInfo{" +
                "createTime=" + createTime +
                ", startTime=" + startTime +
                ", taskMode=" + taskMode +
                ", token='" + token + '\'' +
                ", targetPoint='" + targetPoint + '\'' +
                '}'
    }
}

